provider/datagen/src/transform/cldr/datetime/week_data.rs

```diff
@@ -131,3 +133,193 @@ fn basic_cldr_week_data() {
         gg_week_data.get().first_weekday
     );
 }
+
+impl DataProvider<WeekDataV2Marker> for crate::DatagenProvider {
+    fn load(&self, req: DataRequest) -> Result<DataResponse<WeekDataV2Marker>, DataError> {
+        self.check_req::<WeekDataV2Marker>(req)?;
+        let territory = req
+            .locale
+            .region()
+            .map(|v| -> Result<Territory, DataError> { Ok(Territory::Region(v)) })
+            .transpose()?
+            .unwrap_or_else(|| DEFAULT_TERRITORY.clone());
```

Comment:
`nit`: something is off in this chain. Your `map` is infallible, so why `Result`+`transpose`+`unwrap`?

Reply:
Thank you, addressed in https://github.com/unicode-org/icu4x/pull/4364/commits/a865f0414a74c8a5328c86a3e79a552018927eb1.

Note that the existing V1 does this as well, let me know if you think we should change it in this PR
https://github.com/unicode-org/icu4x/blob/68ac7444a8a127facb1d51c91f57dac07efe9ecc/provider/datagen/src/transform/cldr/datetime/week_data.rs#L44-L46